### PR TITLE
Upgraded library versions to avoid security vulnerability

### PR DESCRIPTION
**Pull request by lijamie98**
<details>
  <summary>PR Checklist</summary>

### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `paymentservice.stellar`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
</details>

### What

The following dependencies have some vulnerabilities and should be updated:

- h2database 1.4.200 → 2.1.214
- kotlin 1.6.20 → 1.7.20
- postgresql 42.4.1 → 42.5.0
- log4j and log4j-template-json 2.17.1 → 2.19.0
- jackson-dataformat-yaml 2.13.2 → 2.14.0
- spring-beans → org.springframework:spring-beans:5.3.20
- tomcat-embed-core: → org.apache.tomcat.embed:tomcat-embed-core: 9.0.63
- kotlin-scripting-jvm → org.jetbrains.kotlin:kotlin-scripting-jvm: 1.7.20
- spring-context → org.springframework:spring-context:5.3.20
- netty-common → io.netty:netty-common:4.1.77.Final
- aws-java-sdk-s3 → com.amazonaws:aws-java-sdk-core:1.12.342
- spring-expression → org.springframework:spring-expression:5.3.20
- spring-messaging → org.springframework:spring-messaging:5.3.20

### Why

Fix security vulnerability.

### Known limitations

N/A

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -53,7 +53,7 @@ subprojects {
       googleJavaFormat()
     }
 
-    kotlin { ktfmt("0.30").googleStyle() }
+    kotlin { ktfmt("0.41").googleStyle() }
```
Comment (lijamie98) [inline build.gradle.kts:56]: This is the result of upgrading to Kotlin 1.7.20

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -13,9 +13,9 @@ findbugs-jsr305 = "3.0.2"
 flyway-core = "8.5.13"
```
Comment (lijamie98) [inline gradle/libs.versions.toml:13]: Please review this file closely.

File: integration-tests/build.gradle.kts [hunk 3]
```diff
@@ -18,6 +18,7 @@ dependencies {
   implementation(libs.java.stellar.sdk)
   implementation(libs.google.gson)
   implementation(libs.okhttp3)
+  implementation(libs.log4j2.api)
```
Comment (lijamie98) [inline integration-tests/build.gradle.kts:21]: `ServiceLoaderUtil` was moved from `log4j2.core` to `log4j2.api` in version `2.19.0`

File: build.gradle.kts (hunk 1 above)
Comment (lijamie98) [reply build.gradle.kts:56]: This also impacts the result of Kotlin formatter which causes a lot of file modifications in this PR.

**Review by C0x41lch0x41 — APPROVED** [review]
Looks good to me. Thanks!

**Pull request merged**